ldes-server-infra-postgres/postgres-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/projection/MemberPropertyVersionProjection.java

```diff
@@ -0,0 +1,24 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.retention.projection;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.entity.MemberPropertiesEntity;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public interface MemberPropertyVersionProjection {
+	String getId();
+	String getCollectionName();
+	String getVersionOf();
+	LocalDateTime getTimestamp();
+	Set<MemberViewsProjection> getViews();
+	Integer getVersionNumber();
+
+	default MemberPropertiesEntity toMemberPropertiesEntity() {
+		return new MemberPropertiesEntity(getId(), getCollectionName(),
+				getViews().stream()
+						.map(MemberViewsProjection::getView)
+						.collect(Collectors.toSet()),
+				getVersionOf(), getTimestamp());
+	}
```

Comment:
Could this be replaced to a mapper class or something, as default methods in interfaces are a bad practice and should only be used when dealing with legacy code